### ISSUE TITLE
Is it possible to create a Grpc server that has http2 transport(via Netty) as well as in process transport.

### ISSUE THREAD

**Issue by aroenK** [grpc/grpc-java#9794]
Hello grpc-java team.

Is it possible to create one Grpc server that has Netty available forPort(xxxx), and has the inprocess transport (forname(uniquename)) available?

I Have a use case where I need regular client connections over the network to grpc, as well as an in-process channel to the grpc server. The in-process channel acts as a passtrough for messages over AMQP.

I am not able to combine the current NettyServerBuilder and InProcessServerBuilder to make one grpc server. For the moment I am creating 2 separate servers that have the same services and interceptors registered.

Is there a way to make this through the current API's?

**Comment by sanjaypujare** [comment]
> For the moment I am creating 2 separate servers that have the same services and interceptors registered.

If that's working why do you need to have one "grpc server" ? Is there anything you are unable to do with this?

**Comment by aroenK** [comment]
It does require some resources running a second server(running on an embedded device). It also increases maintenance because each new service needs to be registered twice as well as any Interceptors.

Cant really say that I am too happy with the current design of having setted up multiple grpc server instances for enabling different transports.

**Comment by sanjaypujare** [comment]
I have not seen anyone else asking for multi-transport server instances and I doubt it will help you save resources. That one server could end up using the same amount of resources. So the current split looks much better- the transport is abstracted out in the server (builder) so a use-case such as yours can always be achieved by using multiple servers with a single service definition.

**Comment by ejona86** [comment]
> It does require some resources running a second server(running on an embedded device).

There used to be some goo wrapping Binder and InProcess to allow using a single server (on Android). But it didn't really provide any advantage and there is essentially no resource difference to using two servers, so was removed. The primary cost of a server is the transport.

"Embedded" means very different things to different people, but if your resources are anything close to Android, we've already looked into it and there's limited benefit to a single server.

> It also increases maintenance because each new service needs to be registered twice as well as any Interceptors.

I'd suggest using a helper method that is passed a ServerBuilder.
